scripts/simple-scylla-cluster-manager.py

```diff
@@ -0,0 +1,776 @@
+        supervisor_process = subprocess.Popen(
+                [sys.executable, os.path.abspath(argv0), "supervisor"],
+                stdin = subprocess.DEVNULL,
+                stdout = supervisor_log,
+                stderr = subprocess.STDOUT,
+                cwd = os.path.abspath(cluster_directory))
+
+    supervisor_client = SupervisorClient(cluster_directory)
+
+    # The supervisor can only successfully open the pipes if they are also
+    # opened here with the opposite mode. Send a "check" command to coordinate
+    # opening the pipes. This command also serves the purpose of checking if the
+    # supervisor was successfully initialized.
+    supervisor_client.send_command(["check"])
+
+    return supervisor_client
+
+
+def _maybe_start_nodes(nodes, start_all, supervisor_client):
+    if not start_all or nodes == 0:
+        return
+
+    for i in range(1, nodes + 1):
+        supervisor_client.send_command(["start-node", f"-i{i}"])
+
+
+def _execute_client_command(argv0, remaining_argv):
+    app_name = os.path.basename(argv0)
+    args = _parse_args(app_name, remaining_argv)
+
+    operation = args.operation
+    args_dict = vars(args)
+    del args_dict["operation"]
+
+    if operation == "create-cluster":
+        os.symlink(os.path.abspath(argv0), os.path.abspath(app_name))
+
+        if args.network_prefix is None:
+            # Avoid picking 0, to not clash with a scylla started with the default parameters
+            network_prefix = random.randint(1, 255)
+        else:
+            network_prefix = args.network_prefix
+
+        if args.path is None:
+            path = os.path.abspath(os.path.join(os.path.dirname(argv0), '..'))
+        else:
+            path = os.path.abspath(args.path)
+
+        config = {
+            "name": args.name,
+            "network_prefix": network_prefix,
+            "default_scylla_config": {
+                "extra_args": args.extra_args,
+                "memory": args.memory,
+                "mode": args.mode,
+                "path": path,
+                "smp": args.smp,
+            },
+            "nodes": { }
+        }
+
+        with open(os.path.join(args.cluster_directory, CONFIG_FILE), "w") as f:
+            json.dump(config, f, indent=4)
+
+        supervisor_client = _start_supervisor(argv0, args.cluster_directory)
+
+        for _ in range(args.nodes):
+            supervisor_client.send_command(['add-node'])
+
+        _maybe_start_nodes(args.nodes, args.all, supervisor_client)
+    elif operation == "start-cluster":
+        supervisor_client = _start_supervisor(argv0, args.cluster_directory)
+        if args.all:
+            nodes_status = json.loads(supervisor_client.send_command(['status']))
+            _maybe_start_nodes(len(nodes_status), args.all, supervisor_client)
+    elif operation == "stop-cluster":
+        supervisor_client = SupervisorClient(args.cluster_directory)
+        supervisor_client.send_command(['stop'])
```

Comment:
might want to forward the return val of `send_command()` to the caller?

Reply:
There is no return value for the `stop` command. It also cannot fail.

Reply:
right. but `send_command()` (should) always returns. i think it be simpler to have less exceptions in the code in the sense that we always return the return value instead of doing this case-by-case. see also https://github.com/scylladb/scylladb/pull/14998/files#r1288732626

Reply:
`send_command()` is like an RPC call and only a single command has return value. All the rest return None.
Adding a `return` here will do nothing for consistency, above we have loops of `send_command()` where the return value is dropped because, there is no way to combine them and no point in trying, we know they don't return anything (failure is converted to exceptions).

Reply:
i see. thank you for the explanations! so the one which returns value is an exception. i thought the other way around. just wasn't quite in favor of exceptions =)

Reply:
I made it more uniform in v2 (still WIP).